### PR TITLE
Graph break on ZeroTensor inputs in Dynamo instead of segfaulting

### LINKED ISSUE #197067: `torch.compile` SEGFAULTs on a ZeroTensor produced by a normal autograd path (gradient of `torch.sgn`)
### 🐛 Describe the bug

A `ZeroTensor` produced by a **normal autograd path** — the gradient of `torch.sgn` — makes a
`torch.compile`d function **segfault** when the tensor is used (indexed) inside it. Eager runs
fine and returns the correct value. This is memory-unsafe (SIGSEGV in the generated kernel),
not a graceful error.

## Minimal repro

```python
import torch

# A ZeroTensor arises naturally as the gradient of sgn (derivative of sgn is 0):
x = torch.randn(3, requires_grad=True)
z = torch.autograd.grad(torch.sgn(x).sum(), x)[0]
assert z._is_zerotensor()          # True -- an efficient ZeroTensor, from plain autograd

def fn(t):
    return t[-1] + 10

print(fn(z))                                  # eager:    tensor(10.)
torch.compile(fn, backend="inductor")(z)      # compiled: SEGFAULT (SIGSEGV, exit 139)
```

Deterministic across runs on **CPU** (the inductor CPU C++ kernel segfaults). On CUDA the
compiled path does not segfault (it completes), so this is a CPU-inductor issue.

## Expected behavior

Compiling a function that reads a ZeroTensor should behave like eager (a ZeroTensor reads as
zeros), or raise a graceful error — it must not segfault. A compiler should never turn a valid
eager program into a memory-unsafe crash.

## Actual behavior

Eager returns `tensor(10.)`. Under `torch.compile` (inductor), the generated kernel reads the
ZeroTensor's absent data pointer and the process crashes with SIGSEGV. So a value produced by
ordinary autograd (`grad` of `sgn`), passed into a compiled function, crashes the process.

### Versions

```
PyTorch version: 2.15.0a0+gite67e43b
Is debug build: True
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 12.3.0-1ubuntu1~22.04.3) 12.3.0
Clang version: 17.0.6
CMake version: version 4.0.0
Libc version: glibc-2.35

Python version: 3.12.13 | packaged by conda-forge (64-bit runtime)
Python platform: Linux-6.8.0-138-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.6.20
GPU 0: NVIDIA RTX A6000
Nvidia driver version: 580.82.07
cuDNN version: 9.3.0

[pip3] numpy==2.3.5
[pip3] torch==2.15.0a0+gite67e43b
[pip3] triton==3.7.1
```

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by kodlan**
Fixes #197067

A ZeroTensor (what autograd returns for the gradient of torch.sgn, for example) has no storage behind it. Dynamo accepted one as a normal graph input, the fake tensor lost the ZeroTensor property, and the Inductor CPU kernel then read through the null data pointer and crashed the process with SIGSEGV. Eager handles the same call fine through the ZeroTensor fallback.

This adds a graph break in wrap_tensor for inputs with _is_zerotensor(), same idea as the dual tensor input check from #189644. It sits at the top of wrap_tensor, above the register_attr_or_module fast paths, because with install_free_tensors or mark_static_address(guard=True) those paths bake the tensor into the graph and would still crash. The frame falls back to eager, which is correct for these tensors, and fullgraph=True gives an Unsupported error instead of a crash. Since the break happens before anything is traced, the frame is skipped, so later calls with dense tensors also run eager until the next reset; that matches the dual tensor case. Registry entry added via the GB_REGISTRY linter. Regression test in test/dynamo/test_repros.py checks the eager fallback with a CompileCounter, the inductor path on CPU, and the fullgraph error; it fails before the change at the frame_count assert.

This is Dynamo only. Non-strict export plus AOTI with a ZeroTensor input still hits the same null read; a complete fix would have to carry the ZeroTensor property through fake tensor conversion. Also not covered here: ZeroTensors created inside a compiled region already lower fine through _efficientzerotensor, and there is a separate pre-existing issue where the C++ add_zerotensor kernel breaks under a TorchDispatchMode because the wrapped scalar arrives as a Python int (that is what aot_eager was hitting).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Review by guilhermeleobas — APPROVED** [review]
The patch looks good to me. But I'm wondering if this error is caused by AOT autograd or Inductor and not Dynamo.

**Comment by guilhermeleobas** [comment]
@kodlan feel free to merge once CI passes

**Review by eellison — COMMENTED** [review]
Could we decompose this instead ?
